### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by lxgr-linux**
This documents methods and functions in the following scheme:
```
ARGS:
    args1: description
    ...
RETURNS:
    blablabla
```
logging should be merged first

### CODE REVIEW

File: pokete.py [hunk 1]
```diff
@@ -975,7 +1084,12 @@ def __call__(self):
         self.box.remove_c_obs()
 
     def rem_item(self, name, items):
-        """Removes an item to the inv"""
+        """Removes an item to the inv
```
Comment (MaFeLP) [inline pokete.py:1087]:
```suggestion
        """Removes an item from the inv
```

File: pokete.py [hunk 2]
```diff
@@ -681,11 +765,18 @@ def give_item(self, item, amount=1):
         logging.info("[Figure] %d %s(s) given", amount, item)
 
     def has_item(self, item):
-        """Checks if an item is already present"""
+        """Checks if an item is already present
+        ARGS:
+            item: Generic item name
+        RETURNS:
+            bool"""
```
Comment (MaFeLP) [inline pokete.py:772]: All other methods say what they return and not the type:

```suggestion
            If the player has this item"""
```

File: pokete.py [hunk 3]
```diff
@@ -726,7 +821,7 @@ class Debug:
 
     @classmethod
     def pos(cls):
-        """Prints the position"""
+        """Prints the figures position"""
```
Comment (MaFeLP) [inline pokete.py:824]:
```suggestion
        """Prints the figures' position"""
```

File: pokete_classes/attack.py [hunk 4]
```diff
@@ -8,7 +8,10 @@
 
 
 class Attack:
-    """Attack that can be used by a Pokete"""
+    """Attack that can be used by a Pokete
+    ARGS:
+        index: The attacks basic name
+        p_data: P_data module"""
```
Comment (MaFeLP) [inline pokete_classes/attack.py:14]:
```suggestion
        p_data: p_data module"""
```

File: pokete_classes/attack_actions.py [hunk 5]
```diff
@@ -1,5 +1,10 @@
 class AttackActions:
-    """This class contains all actions callable by an attack"""
+    """This class contains all actions callable by an attack
+    All this methods belong to one or more attacks and follow the following
```
Comment (MaFeLP) [inline pokete_classes/attack_actions.py:3]:
```suggestion
    All these methods belong to one or more attacks and follow the following
```

File: pokete_classes/attack_actions.py [hunk 6]
```diff
@@ -1,5 +1,10 @@
...
+    patern:
```
Comment (MaFeLP) [inline pokete_classes/attack_actions.py:4]:
```suggestion
    pattern:
```

File: pokete_classes/classes.py [hunk 7]
```diff
@@ -50,12 +61,16 @@ class OutP(se.Text):
     """Output label to better organize output"""
 
     def outp(self, text):
-        """Rechar and show wrapper"""
+        """Rechar and show wrapper
+        ARGS:
+            text: String thats printed out"""
```
Comment (MaFeLP) [inline pokete_classes/classes.py:66]:
```suggestion
            text: String that's printed out"""
```

File: pokete_classes/detail.py [hunk 8]
```diff
@@ -12,7 +12,14 @@ class Informer:
 
     @staticmethod
     def add(poke, figure, _map, _x, _y, in_deck=True):
-        """Adds a Pokete to the deck"""
+        """Adds a Poketes info to the deck
```
Comment (MaFeLP) [inline pokete_classes/detail.py:15]: Not sure about that one, but I think there is an apostrophe missing:
```suggestion
        """Adds a Pokete's info to the deck
```

File: pokete_classes/detail.py [hunk 9]
```diff
@@ -73,7 +85,11 @@ def __init__(self, height, width):
         self.line_middle.add(self.map, round(self.map.width / 2), 7)
 
     def __call__(self, _ev, poke, abb=True):
-        """Shows details"""
+        """Shows details
+        ARGS:
+            _ev: Event object
+            poke: Poke object thats details are given
```
Comment (MaFeLP) [inline pokete_classes/detail.py:91]:
```suggestion
            poke: Poke object whose details are given
```

File: pokete_classes/detail.py [hunk 10]
```diff
@@ -73,7 +85,11 @@ def __init__(self, height, width):
...
+            abb: Bool whether or not the abbility option is shown"""
```
Comment (MaFeLP) [inline pokete_classes/detail.py:92]:
```suggestion
            abb: Bool whether or not the ability option is shown"""
```

File: pokete_classes/effects.py [hunk 11]
```diff
@@ -5,8 +5,24 @@
 
 
 class Effect():
+    """An effect that can be given to a Pokete and that effects the Pokete
+    during fights
+    ARGS:
+        name: The effects displayed name (paralysed etc)
+        rem_chance: The chance the effect gets removed
+        catch_chance: The number with which the catch chnace of the Poke is
```
Comment (MaFeLP) [inline pokete_classes/effects.py:13]:
```suggestion
        catch_chance: The number with which the catch chance of the Poke is
```

File: pokete_classes/effects.py [hunk 12]
```diff
@@ -5,8 +5,24 @@
...
+            increased
+        text: The text shown near the Pokes name ("(Bur)")
+        str_esccode: The color of sayed label,
```
Comment (MaFeLP) [inline pokete_classes/effects.py:16]:
```suggestion
        str_esccode: The color of said label,
```

File: pokete_classes/effects.py [hunk 13]
```diff
@@ -5,8 +5,24 @@
...
+        obj: The Poke the effect is added to
+        exclude: A list of type names that the effect can't be added to
+    CLASS VARS:
+        desc: The effects discription
```
Comment (MaFeLP) [inline pokete_classes/effects.py:20]:
```suggestion
        desc: The effects description
```

File: pokete_classes/effects.py [hunk 14]
```diff
@@ -5,8 +5,24 @@
...
+        c_name: The classes simplified name
```
Comment (MaFeLP) [inline pokete_classes/effects.py:21]:
```suggestion
        c_name: The class' simplified name
```

File: pokete_classes/effects.py [hunk 15]
```diff
@@ -25,6 +41,9 @@ def __repr__(self):
         return f"{type(self).__name__}"
 
     def add(self, obj):
+        """Adds the effect to a Pokete
+        ARGS:
+            obj: The Poke the effect is adde to"""
```
Comment (MaFeLP) [inline pokete_classes/effects.py:46]:
```suggestion
            obj: The Poke the effect is added to"""
```

File: pokete_classes/fightmap.py [hunk 16]
```diff
@@ -122,15 +146,22 @@ def add_2(self, player):
...
             arr[_i].add(self, setob.x, setob.y)
             self.show()
             time.sleep(0.1)
 
     def get_attack(self, _ev, attack_obs):
-        """Inputloop for attack options"""
+        """Inputloop for attack options
+        ARGS:
+            _ev: Event object
+            attack_obs: A list of Attack objects taht belong to a Poke"""
```
Comment (MaFeLP) [inline pokete_classes/fightmap.py:164]:
```suggestion
            attack_obs: A list of Attack objects that belong to a Poke"""
```

File: pokete_classes/fightmap.py [hunk 17]
```diff
@@ -178,7 +213,24 @@ def get_item(self, _ev, items, inv):
 
     def fight(self, player, enemy, figure, settings, invitems, fightitems,
               deck, p_data, _ev, info):
-        """Fight"""
+        """Fight between two Pokes
+        ARGS:
+            player: The players used Poke
```
Comment (MaFeLP) [inline pokete_classes/fightmap.py:218]:
```suggestion
            player: The players' used Poke
```

File: pokete_classes/fightmap.py [hunk 18]
```diff
@@ -178,7 +213,24 @@ def get_item(self, _ev, items, inv):
...
+            enemy: The enemys used Poke
```
Comment (MaFeLP) [inline pokete_classes/fightmap.py:219]:
```suggestion
            enemy: The enemy's used Poke
```

File: pokete_classes/fightmap.py [hunk 19]
```diff
@@ -334,20 +383,50 @@ def fight(self, player, enemy, figure, settings, invitems, fightitems,
...
 
-    def __init__(self, _map, movemap, figure, ob_maps):
+    The methods that can actually be called in fight follow the follwing patern:
+        ARGS:
+            obj: The players Poke object
+            enem: The enemys Poke object
+            info: The info dict
+        RETURNS:
+            1: The continue the attack round
+            2: The win the game
```
Comment (MaFeLP) [inline pokete_classes/fightmap.py:407]: What is meant with `The continue the attack round`?

Maybe try replacing one `the`?

File: pokete_classes/input.py [hunk 20]
```diff
@@ -60,15 +71,27 @@ def ask_bool(_ev, _map, text):
...
 def ask_text(_ev, _map, infotext, introtext, text, name, max_len):
-    """Asks the player to input a text"""
+    """Asks the player to input a text
+    ARGS:
+        _ev: Event object
+        _map: The map the input box should be shown on
+        infotext: The information text about the input
+        introtext: The text that introduces the text field
+        text: The default text in the text field
+        name: The boxes desplayed name
```
Comment (MaFeLP) [inline pokete_classes/input.py:81]:
```suggestion
        name: The boxes displayed name
```

File: pokete_classes/inv_items.py [hunk 21]
```diff
@@ -15,7 +17,13 @@ def __init__(self, p_d):
 
 
 class InvItem:
-    """Item for the inventory"""
+    """Item for the inventory
+    ARGS:
+        name: The items generic name (healing_potion)
+        pretty_name: The items pretty name (Healing potion)
+        desc: The items description
+        price: The items price in the shop
```
Comment (MaFeLP) [inline pokete_classes/inv_items.py:25]:
```suggestion
        name: The item's generic name (healing_potion)
        pretty_name: The item's pretty name (Healing potion)
        desc: The item's description
        price: The item's price in the shop
```

File: pokete_classes/inv_items.py [hunk 22]
```diff
@@ -26,7 +34,10 @@ def __init__(self, name, pretty_name, desc, price, fn=None):
 
 
 class LearnDisc(InvItem):
-    """Learning disc item to teach attacks to Poketes"""
+    """Learning disc item to teach attacks to Poketes
+    ARGS:
+        attack_name: The nama of the attack beeing taught
```
Comment (MaFeLP) [inline pokete_classes/inv_items.py:39]:
```suggestion
        attack_name: The name of the attack being taught
```

File: pokete_classes/learnattack.py [hunk 23]
```diff
@@ -12,7 +12,11 @@
 
 
 class AttackInfo(Box):
-    """Gives information about a certain attack"""
+    """Gives information about a certain attack
+    ARGS:
+        attack: The attacks name
```
Comment (MaFeLP) [inline pokete_classes/learnattack.py:17]:
```suggestion
        attack: The attack's name
```

File: pokete_classes/learnattack.py [hunk 24]
```diff
@@ -36,15 +40,26 @@ def __enter__(self):
...
         self.box = ChooseBox(6, 25, name="Attacks", info="1:Details, 2:Info")
 
     def __call__(self, _ev, p_data, attack=None):
-        """Starts the learning process"""
+        """Starts the learning process
+        ARGS:
+            _ev: Event object
+            p_data: p_data module
+            attack: The attacks' name that should be leared, if None a fitting
+                    attack will be choosen randomly
```
Comment (MaFeLP) [inline pokete_classes/learnattack.py:59]:
```suggestion
            attack: The attack's name that should be learned, if None a fitting
                    attack will be chosen randomly
```

File: pokete_classes/mods.py [hunk 25]
```diff
@@ -15,8 +15,10 @@ def __init__(self):
 
 class ModError(Exception):
     """
-    An Error that is thrown, when an inproper module is loaded.
-    """
+    An Error that is thrown, when an inproper module is loaded
+    ARGS:
+        name: The mods name
```
Comment (MaFeLP) [inline pokete_classes/mods.py:20]:
```suggestion
        name: The mod's name
```

File: pokete_classes/movemap.py [hunk 26]
```diff
@@ -33,7 +37,12 @@ def add_obs(self):
         self.label.add(self, 0, self.height - 1)
 
     def text(self, x, y, inp_arr, _ev):
-        """Shows dialog text on movemap"""
+        """Shows dialog text on movemap
+        ARGS:
+            x: The messages X
```
Comment (MaFeLP) [inline pokete_classes/movemap.py:42]:
```suggestion
            x: The message's X
```

File: pokete_classes/movemap.py [hunk 27]
```diff
@@ -77,15 +87,19 @@ def resize(self, height, width, background=" "):
         self.add_obs()
 
     def balls_label_rechar(self, pokes):
-        """Rechars the balls label"""
+        """Rechars the balls label
+        ARGS:
+            pokes: The players Pokes"""
```
Comment (MaFeLP) [inline pokete_classes/movemap.py:92]:
```suggestion
            pokes: The player's Pokes"""
```

File: pokete_classes/movemap.py [hunk 28]
```diff
@@ -77,15 +87,19 @@ def resize(self, height, width, background=" "):
         self.add_obs()
 
     def balls_label_rechar(self, pokes):
-        """Rechars the balls label"""
+        """Rechars the balls label
```
Comment (MaFeLP) [inline pokete_classes/movemap.py:90]:
```suggestion
        """Rechars the ball's label
```

File: pokete_classes/movemap.py [hunk 29]
```diff
@@ -77,15 +87,19 @@ def resize(self, height, width, background=" "):
...
                                 or pokes[i].identifier == "__fallback__"\
                                         else "o" if pokes[i].hp > 0
                                         else "x"
                                         for i in range(6)), esccode=Color.thicc)
 
     def name_label_rechar(self, name):
-        """Rechars name_label and sets balls_label correctly"""
+        """Rechars name_label and sets balls_label correctly
+        ARGS:
+            name: The players new name"""
```
Comment (MaFeLP) [inline pokete_classes/movemap.py:102]:
```suggestion
            name: The player's new name"""
```

File: pokete_classes/roadmap.py [hunk 30]
```diff
@@ -7,7 +8,16 @@
...
 
 class Station(se.Square):
-    """Selectable station for Roadmap"""
+    """Selectable station for Roadmap
+    ARGS:
+        roadmap: RoadMap object
+        associate: Main PlayMap name the station belongs to
+        additionals: List of PlayMap names the station also belongs to
+        width: The Stations width
+        height: The Stations height
```
Comment (MaFeLP) [inline pokete_classes/roadmap.py:17]:
```suggestion
        width: The Station's width
        height: The Station's height
```

File: pokete_classes/roadmap.py [hunk 31]
```diff
@@ -7,7 +8,16 @@
...
+        desc: The associates description
```
Comment (MaFeLP) [inline pokete_classes/roadmap.py:18]:
```suggestion
        desc: The associated description
```

File: pokete_classes/roadmap.py [hunk 32]
```diff
@@ -7,7 +8,16 @@
...
+        char: Displayed char
+        {w,a,s,d}_next: The next Stations name in a certain direction"""
```
Comment (MaFeLP) [inline pokete_classes/roadmap.py:20]:
```suggestion
        {w,a,s,d}_next: The next Station's name in a certain direction"""
```

File: pokete_data/__init__.py [hunk 33]
```diff
@@ -1,6 +1,7 @@
 """This provides p_data. Never ever import this except for in pokete.py since
    p_data can be manipulated by mods and therefore should be injected and not
-   imported"""
+   imported
+   I know all this is very awfull..."""
```
Comment (MaFeLP) [inline pokete_data/__init__.py:4]:
```suggestion
   imported

   I know all this is very awfull..."""
```

File: pokete_data/__init__.py [hunk 34]
```diff
@@ -14,22 +15,34 @@
 
 
 class ValidationError(Exception):
+    """Error thrown when validation fails"""
```
Comment (MaFeLP) [inline pokete_data/__init__.py:18]: As you explained nicely in the other classes, what are the parameters for?

File: pokete_general_use_fns.py [hunk 35]
```diff
@@ -31,19 +42,27 @@ def hard_liner(l_len, name):
...
+        _ev: Event object"""
+    if _ev.get() == "exit":
         raise KeyboardInterrupt
 
 
 def print_help(path):
-    """Shows help message"""
+    """Shows help message
+    ARGS:
+        path: The games path"""
```
Comment (MaFeLP) [inline pokete_general_use_fns.py:65]:
```suggestion
        path: The game's path"""
```

**Review by MaFeLP — CHANGES_REQUESTED** [review]

File: pokete_classes/fightmap.py (hunk 19 above)
Comment (lxgr-linux) [reply pokete_classes/fightmap.py:407]: It should have been a `to`

**Review by MaFeLP — APPROVED** [review]

**Pull request merged**